src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,252 @@
+        //java.nio.file.Files.exists(path)
+        try {
+            File f = new File(path.toString());
+            if (f.createNewFile()) {
+                System.out.println("File created: " + path.toAbsolutePath());
+            } else {
+                System.out.println("File already exists.\nLoading...");
+                Scanner s = new Scanner(f); // create a Scanner using the File as the source
+                while (s.hasNext()) {
+                    String[] lineParts = s.nextLine().split(FILE_SEPARATOR);
+                    switch(lineParts[0]) {
+                    case "Todo":
+                        try {
+                            addTask(new Todo(lineParts[2], Boolean.parseBoolean(lineParts[1])));
+                        } catch (DukeException e) {
+                            System.out.print("The description of a todo cannot be empty.\n");
+                        }
+                        break;
+                    case "Deadline":
+                        try {
+                            addTask(new Deadline(lineParts[2], lineParts[3], Boolean.parseBoolean(lineParts[1])));
+                        } catch (IndexOutOfBoundsException e) {
+                            System.out.print("Something went wrong. Please put the due date after /by.\n");
+                        } catch (DukeException e) {
+                            System.out.print("The description of a deadline cannot be empty.\n");
+                        }
+                        break;
+                    case "Event":
+                        try {
+                            addTask(new Event(lineParts[2], lineParts[3], Boolean.parseBoolean(lineParts[1])));
+                        } catch (IndexOutOfBoundsException e) {
+                            System.out.print("Something went wrong. Please put the event time after /at.\n");
+                        } catch (DukeException e) {
+                            System.out.print("The description of a event cannot be empty.\n");
+                        }
+                        break;
+                    }
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred.");
+            e.printStackTrace();
+        }
+    }
+
+    private static void writeToFile(Path path) throws IOException {
+        FileWriter fw = new FileWriter(path.toString());
+        for (Task t: tasks) {
+            if("Todo".equals(t.getTaskType())) {
+                fw.write(t.getTaskType() + FILE_SEPARATOR + t.getDone() + FILE_SEPARATOR
+                        + t.getDescription() + "\n");
+            } else {
+                fw.write(t.getTaskType() + FILE_SEPARATOR + t.getDone() + FILE_SEPARATOR
+                        + t.getDescription() + FILE_SEPARATOR  + t.getDate() + "\n");
+            }
+        }
+        fw.close();
+    }
+
+    public static void main(String[] args) {
+        String home = System.getProperty("user.home");
```

Comment:
`println` statements alternate between methods. Could you consider any way if [SLAP](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) could be maintained?

Reply:
I have corrected some of it, but similar to the previous comment, I am unsure if I am suppose to handle the exceptions within the methods